### PR TITLE
feat(workspace): add workspace settings management tools

### LINKED ISSUE #80: Expand workspace management tools
## Problem

Only `list_workspaces` is currently implemented.
Workspace access control, authentication settings, notification preferences, user preferences, MFA methods, and usage/billing information are all missing.
MFA method listing is especially important in remote mode where it connects to Auth0 MFA configuration.

## Current state

| CLI command | MCP tool | Status |
|-------------|----------|--------|
| `workspace list` | `list_workspaces` | ✅ |
| `workspace access-control` | — | ❌ |
| `workspace access-control update` | — | ❌ |
| `workspace authentication` | — | ❌ |
| `workspace authentication update` | — | ❌ |
| `workspace notifications` | — | ❌ |
| `workspace notifications update` | — | ❌ |
| `workspace preferences` | — | ❌ |
| `workspace preferences update` | — | ❌ |
| `workspace mfa-methods` | — | ❌ |
| `workspace usage` | — | ❌ |

## Target API endpoints

| Endpoint | Method | Description |
|----------|--------|-------------|
| `/api/workspaces/access-control/-/` | GET/PATCH | Access control settings |
| `/api/workspaces/security/-/` | GET/PATCH | Authentication/security settings |
| `/api/workspaces/security/-/mfa-methods/` | GET | MFA methods list |
| `/api/workspaces/notifications/` | GET/PATCH | Notification preferences |
| `/api/workspaces/preferences/-/` | GET/PATCH | User preferences |
| `/api/workspaces/workspaces/` | GET | Usage/billing information |

## MCP tools to implement

```python
# tools/workspace_tools.py extension

get_workspace_access_control(workspace, region)
update_workspace_access_control(workspace, region, ...)
get_workspace_security(workspace, region)
update_workspace_security(workspace, region, ...)
list_workspace_mfa_methods(workspace, region)
get_workspace_notifications(workspace, region)
update_workspace_notifications(workspace, region, ...)
get_workspace_preferences(workspace, region)
update_workspace_preferences(workspace, region, ...)
get_workspace_usage(workspace, region)
```

## Acceptance criteria

- [ ] Add missing tools to `tools/workspace_tools.py`
- [ ] Register new tools in `server.py`
- [ ] Verify MFA methods tool works correctly in remote mode
- [ ] Add unit tests (`tests/test_workspace_tools.py`)

### `list_workspace_mfa_methods` mode-specific behavior

- **streamable-http**: Directly connected to Auth0 MFA configuration. Useful for proactively informing users of available MFA methods when a re-authentication flow is triggered (see `utils/auth_error_middleware.py`). Primary value is in this mode.
- **stdio**: API token-based auth means MFA re-authentication never occurs. Meaningful only as a read-only lookup.

### `update_workspace_security` caveat

Strengthening MFA policy may immediately trigger a `401 + mfa` pseudo-scope response on the current remote mode session, requiring re-authentication. Recommend including a warning in the tool description: "Changes to security policy may require re-authentication in the current session."

**Affected files**: `tools/workspace_tools.py`
**Related docs**: `docs/gap-analysis-2026-04-30.md` §5.5

### PR DESCRIPTION
**Pull request by jisung-02**
## Background

The only workspace-related MCP tools so far were `list_workspaces` and `get_current_user`. Alpacon's `/api/workspaces/` surface exposes management endpoints for access control, security/authentication settings, the allowed MFA methods, notification settings, and workspace-wide preferences, but none of them were reachable through MCP, so an agent could neither inspect a workspace's policy state nor change its general settings. Issue #80's original draft proposed six endpoints and full read/update coverage; the scope was adjusted after checking each claim against the actual server implementation.

## Changes

Added seven workspace-settings tools to `tools/workspace_tools.py`.

Read tools:
- `get_workspace_access_control`: GET `/api/workspaces/access-control/-/`
- `get_workspace_security`: GET `/api/workspaces/security/-/`
- `list_workspace_mfa_methods`: GET `/api/workspaces/security/-/mfa-methods/`
- `get_workspace_notifications`: GET `/api/workspaces/notifications/-/`
- `get_workspace_preferences`: GET `/api/workspaces/preferences/-/`

Update tools (partial update; only provided fields are sent):
- `update_workspace_notifications`: PATCH `/api/workspaces/notifications/-/`
- `update_workspace_preferences`: PATCH `/api/workspaces/preferences/-/`

All tools follow the existing `@mcp_tool_handler` + `unwrap_http_result` + `success_response` pattern, with `READ_ONLY` on the read tools and `IDEMPOTENT_WRITE` on the update tools. The five read tools are also exposed as `alpacon://workspace-settings/{access-control|security|mfa-methods|notifications|preferences}/{region}/{workspace}` resources in `tools/resources.py`. The tool and resource lists in CLAUDE.md, `docs/api-reference.md`, and the `README.md` workspace section were updated, and a `CHANGELOG.md` Unreleased entry was added.

## Out of scope

Some items from the original draft were dropped because they did not match the real server implementation or the platform's policy model.

Security/access-control update tools: the server gates writes to these settings behind a superuser session with fresh MFA re-verification. A static API token (stdio mode) can never satisfy this, and a remote-mode caller would have to complete browser MFA before every change. These are governance-level settings, so only reads are exposed and writes stay human-only via the web console.

Usage/billing lookup: the `/api/workspaces/workspaces/` list response carries no usage data, and the real usage endpoint (`/api/workspaces/workspaces/{id}/usage/`) only accepts portal service-to-service (M2M) credentials, so it cannot be called with a workspace token. Deferred until a customer-facing usage API exists.

Per-user preferences: the draft labeled this "user preferences", but `/api/workspaces/preferences/-/` is workspace-wide configuration. Per-user settings live in a separate app (`/api/profiles/preferences/-/`) and can be handled in a follow-up issue if needed.

## Permissions

| Endpoint | Read | Write |
|---|---|---|
| access-control | exposed | not exposed (superuser + MFA, web console only) |
| security | exposed (SaaS only) | not exposed (superuser + MFA, web console only) |
| mfa-methods | exposed | n/a |
| notifications | exposed | exposed (superuser) |
| preferences | exposed | exposed (admin) |

## Safety

`get_workspace_security` is a SaaS-only route and returns 404 on on-premise deployments; the tool translates that into a clear "not available on this deployment" message instead of a generic upstream error. This 404 branch requires the http_client `error` key (matching `unwrap_http_result`), so a success payload that happens to carry `status_code: 404` is not misread. For `update_workspace_preferences`, the tool description and docs warn that `timezone` is the workspace's billing clock (changing it shifts the daily usage-aggregation boundary), and that `billing_email` / `allowed_domains` are only accepted by the server on SaaS deployments.

## Testing

`tests/test_workspace_tools.py` adds a test class per tool. Read tools cover success and error cases; `get_workspace_security` additionally covers the on-premise 404 to clear-message path, confirms that handling does not leak into other error codes (e.g. 500), and asserts that a success payload carrying `status_code: 404` is not misread as unavailable; update tools verify that partial update sends only the provided fields, that no fields provided returns an error, and the HTTP error envelope. Full suite `pytest tests/ -q` reports 924 passed (7 pre-existing unrelated warnings); `ruff check` / `ruff format` pass.

## Checklist

- [x] Add seven tools and register them via `@mcp.tool`
- [x] Expose `alpacon://` resources for the five read tools
- [x] Update CLAUDE.md, docs/api-reference.md, README.md, and CHANGELOG.md
- [x] Add unit tests and confirm the full suite passes
- [ ] Verify `list_workspace_mfa_methods` against a live remote (streamable-http) session

Closes #80

### CODE REVIEW

File: tools/workspace_tools.py [hunk 1]
```diff
@@ -182,3 +182,409 @@ async def get_current_user(
...
+    token = kwargs.get('token')
+
+    result = await http_client.get(
+        region=region,
+        workspace=workspace,
+        endpoint='/api/workspaces/security/-/',
+        token=token,
+    )
+
+    if isinstance(result, dict) and result.get('status_code') == 404:
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply tools/workspace_tools.py]: Fixed. The 404 special-case now also requires the `error` key so it only fires on an http_client error envelope, matching `unwrap_http_result` (`isinstance(result, dict) and 'error' in result and result.get('status_code') == 404`). Verified this was the only status_code-based 404 check on the branch; the sibling patterns in `server_tools.py`, `metrics_tools.py`, and `unwrap_http_result` already gate on `error`. Added a regression test (`test_success_payload_with_status_code_404_is_not_misread`) covering a success payload that carries `status_code: 404` with no `error` key.

File: tools/workspace_tools.py [hunk 2]
```diff
@@ -182,3 +182,409 @@ async def get_current_user(
...
+
+    Args:
+        workspace: Workspace name. Required parameter
+        front_url: Workspace front-end URL (optional)
+        country: Workspace country code (optional)
+        language: Workspace locale/language code (optional)
+        timezone: Workspace timezone; also the billing clock (optional)
+        invite_ttl: Invitation link time-to-live, in seconds (optional)
+        enabled_extensions: List of enabled extension names (optional)
+        websh_session_timeout: WebSH idle session timeout, in seconds (optional)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply tools/workspace_tools.py]: Fixed to "Websh". Swept the whole repo for the incorrect "WebSH" form; the only two occurrences were this docstring and docs/api-reference.md:579 (both fixed). Everything else already used "Websh".

File: docs/api-reference.md [hunk 3]
```diff
@@ -517,6 +517,74 @@ Get list of available workspaces.
...
+
+**Parameters:**
+- `workspace` (string): Workspace name
+- `front_url` (string, optional): Workspace front-end URL
+- `country` (string, optional): Workspace country code
+- `language` (string, optional): Workspace locale/language code
+- `timezone` (string, optional): Workspace timezone; also the billing clock
+- `invite_ttl` (integer, optional): Invitation link time-to-live, in seconds
+- `enabled_extensions` (array, optional): List of enabled extension names
+- `websh_session_timeout` (integer, optional): WebSH idle session timeout, in seconds
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply docs/api-reference.md]: Fixed to "Websh" to match the project convention in .github/copilot-instructions.md.

File: tools/workspace_tools.py [hunk 4]
```diff
@@ -182,3 +182,413 @@ async def get_current_user(
...
+
+    Returns:
+        Allowed MFA methods response
+    """
+    token = kwargs.get('token')
+
+    result = await http_client.get(
+        region=region,
+        workspace=workspace,
+        endpoint='/api/workspaces/security/-/mfa-methods/',
```
Comment (junho226) [inline tools/workspace_tools.py:359]: Warning: This endpoint is a sub-route of the same `SecuritySettingsViewSet` that backs `get_workspace_security` — on the server it's only registered under `AUTH0_ENABLED` and `initial()` raises `NotFound` otherwise, so it returns 404 on-premise just like `get_workspace_security`. `get_workspace_security` converts that 404 into a clear "SaaS-only, not available on this deployment" message, but this tool falls through to a generic "Not found". Consider applying the same 404 guard here (or factoring it into a small shared helper used by both) so on-premise callers get the actionable message. Non-blocking.

File: tools/workspace_tools.py [hunk 5]
```diff
@@ -182,3 +182,413 @@ async def get_current_user(
...
+    token = kwargs.get('token')
+
+    update_data: dict[str, Any] = {}
+    if disconnection_notification is not None:
+        update_data['disconnection_notification'] = disconnection_notification
+    if notification_channels is not None:
+        update_data['notification_channels'] = notification_channels
+
+    if not update_data:
+        return error_response('No update data provided')
```
Comment (junho226) [inline tools/workspace_tools.py]: Suggestion: Minor consistency — every other error/success response in these tools carries `region`/`workspace`, but this early-return omits them.

```suggestion
        return error_response('No update data provided', region=region, workspace=workspace)
```

File: tools/workspace_tools.py [hunk 6]
```diff
@@ -182,3 +182,413 @@ async def get_current_user(
...
+        update_data['auto_agent_upgrade'] = auto_agent_upgrade
+    if package_proxy is not None:
+        update_data['package_proxy'] = package_proxy
+    if billing_email is not None:
+        update_data['billing_email'] = billing_email
+    if allowed_domains is not None:
+        update_data['allowed_domains'] = allowed_domains
+
+    if not update_data:
+        return error_response('No update data provided')
```
Comment (junho226) [inline tools/workspace_tools.py]: Suggestion: Same as the notifications tool — add `region`/`workspace` here for a uniform response shape.

```suggestion
        return error_response('No update data provided', region=region, workspace=workspace)
```

**Review by junho226 — COMMENTED** [review]
## PR Review: #137 - feat(workspace): add workspace settings management tools

**Author**: @jisung-02
**Branch**: 80-expand-workspace-management-tools → main
**Stack**: python
**Changes**: +917 -16 in 7 files

---

### Summary
Adds seven workspace-settings MCP tools to `tools/workspace_tools.py`: five read tools (`get_workspace_access_control`, `get_workspace_security`, `list_workspace_mfa_methods`, `get_workspace_notifications`, `get_workspace_preferences`) and two partial-update tools (`update_workspace_notifications`, `update_workspace_preferences`). The five read tools are also exposed as `alpacon://workspace-settings/...` resources. Security and access-control writes are deliberately omitted (documented rationale). The change is additive, follows the established `@mcp_tool_handler` + `unwrap_http_result`/`success_response`/`error_response` pattern exactly, and ships with CHANGELOG/README/CLAUDE.md/api-reference docs plus a thorough test suite. Field names were cross-checked against the server serializers and match exactly.

### Critical Issues (0)
None.

### Warnings (1)
- **[tools/workspace_tools.py:327]** `list_workspace_mfa_methods` targets a sub-route of the same `SecuritySettingsViewSet` that backs `get_workspace_security`, so it returns 404 on-premise just like `get_workspace_security` — but `get_workspace_security` converts that into a clear "SaaS-only" message while this tool falls through to a generic "Not found". Same SaaS-only route, asymmetric handling. Recommend applying the same 404 guard here (or a small shared helper). Non-blocking.

### Suggestions (3)
- **[CHANGELOG.md:14 / CLAUDE.md]** The omit-writes rationale ("superuser session with fresh MFA, unsatisfiable by a static token") is exactly true for `security` but only partially for `access-control`, whose on-premise write path gates with `IsAdminUser` + token scope and explicitly skips MFA. The read-only decision is still correct; just soften the one-line justification.
- **[tools/workspace_tools.py:472, 575]** The `error_response('No update data provided')` early-returns omit the `region`/`workspace` context that every other response in these tools carries. Minor consistency nit.
- **[tools/workspace_tools.py:445, 511]** The two update tools share an identical shape — flagged only to pre-empt a "DRY this up" suggestion: the explicit per-field typed parameters are what produce a good MCP input schema, so the duplication is deliberate and correct.

### Highlights
- Deliberate, documented security posture: security/access-control writes are kept human-only via the web console, with the reasoning recorded in the CHANGELOG and CLAUDE.md.
- The SaaS-only 404 guard in `get_workspace_security` correctly checks `'error' in result` **and** `status_code == 404`, and ships a dedicated regression test proving a success payload carrying `status_code: 404` is not misread.
- Field names and SaaS-only fields (`billing_email`, `allowed_domains`) match the server serializers exactly — docs are accurate, not aspirational.
- Partial-update tools are correctly annotated `IDEMPOTENT_WRITE`; read tools `READ_ONLY`; write tools correctly excluded from the resource registry.
- Clean DRY refactor promoting `mock_http_client`/`mock_token` to module-level fixtures, and strong test coverage (success, http-error, partial/multi-field update, empty-update error, and the SaaS 404 edge cases).

---

**Verdict**: COMMENTED

**Checklist**:
- [x] Code follows stack conventions
- [x] Tests included and passing
- [x] No security vulnerabilities
- [x] Documentation updated if needed

File: tools/workspace_tools.py (hunk 4 above)
Comment (jisung-02) [reply tools/workspace_tools.py:359]: Good catch. Applied the same guard and factored it into a shared helper as you suggested. `_saas_only_security_404(result, region, workspace)` now translates the SaaS-only 404 for both `get_workspace_security` and `list_workspace_mfa_methods` into the "not available on this deployment" message. The helper keeps the `unwrap_http_result` contract (requires the `error` key), so a success payload that happens to carry `status_code: 404` is not misread. Added a regression test for the mfa-methods 404 path (and the not-misread case) in cb5272e.

File: tools/workspace_tools.py (hunk 5 above)
Comment (jisung-02) [reply tools/workspace_tools.py]: Applied — the "No update data provided" early-return now passes `region`/`workspace` for a uniform response shape, and the test asserts both fields are present. cb5272e.

File: tools/workspace_tools.py (hunk 6 above)
Comment (jisung-02) [reply tools/workspace_tools.py]: Applied here too, same as the notifications tool. cb5272e.

File: tools/workspace_tools.py [hunk 7]
```diff
@@ -182,3 +209,412 @@ async def get_current_user(
...
+        'API, and this tool reports that the settings are not available on this deployment '
+        'rather than a generic error. '
+        'Related: list_workspace_mfa_methods (allowed methods only), get_workspace_access_control.'
+    ),
+    annotations=READ_ONLY,
+    meta={
+        'anthropic/searchHint': 'workspace security mfa authentication settings',
+    },
+)
+async def get_workspace_security(
```
Comment (junho226) [inline tools/workspace_tools.py:282]: Warning: `get_workspace_security` (and `list_workspace_mfa_methods` below) is effectively JWT-only. Server-side `SecuritySettingsViewSet.authentication_classes` is `[Auth0JWTAuthentication, PortalJWTAuthentication, AccountJWTAuthentication]` (and the `mfa_methods` action uses `Auth0JWTNonMFAAuthentication`) — no `APITokenAuthentication`. In stdio mode (the default) with a token.json API token against a SaaS workspace, upstream returns 403 (AnonymousUser), which slips past `_saas_only_security_404` (403 ≠ 404) and surfaces a raw, confusing error. Consider stacking `@require_jwt_auth` here (same precedent as the token-management tools), or at least documenting the JWT-only constraint.

File: tools/workspace_tools.py [hunk 8]
```diff
@@ -182,3 +209,412 @@ async def get_current_user(
...
+    notification_channels: list[str] | None = None,
+    region: str = '',
+    **kwargs,
+) -> dict[str, Any]:
+    """Update the workspace notification settings (partial update).
+
+    Args:
+        workspace: Workspace name. Required parameter
+        disconnection_notification: Notify when a server disconnects (optional)
+        notification_channels: Channel types to notify through, e.g. email/webhook/push (optional)
```
Comment (junho226) [inline tools/workspace_tools.py]: Warning: `notification_channels` is a full-array replacement on PATCH (server ArrayField), but this description reads as additive. An agent told "also notify via webhook" that sends `notification_channels=["webhook"]` will silently drop email/push. Suggest noting it is replace-in-full — read the current list via `get_workspace_notifications`, merge, then send.

File: tools/workspace_tools.py [hunk 9]
```diff
@@ -182,3 +209,412 @@ async def get_current_user(
...
+    """Update the workspace-wide preferences (partial update).
+
+    Args:
+        workspace: Workspace name. Required parameter
+        front_url: Workspace front-end URL (optional)
+        country: Workspace country code (optional)
+        language: Workspace locale/language code (optional)
+        timezone: Workspace timezone; also the billing clock (optional)
+        invite_ttl: Invitation link time-to-live, in seconds (optional)
+        enabled_extensions: List of enabled extension names (optional)
```
Comment (junho226) [inline tools/workspace_tools.py]: Warning: same full-array-replacement footgun as `notification_channels`. `enabled_extensions` (and `allowed_domains`) replace the whole array on PATCH, and narrowing `enabled_extensions` additionally 402s on non-enterprise plans (`validate_enabled_extensions`). Suggest documenting the replace-in-full semantics (read via `get_workspace_preferences`, merge, then send).

File: CLAUDE.md [hunk 10]
```diff
@@ -480,6 +480,15 @@ All validators are defined in `utils/error_handler.py` and return user-friendly
...
+- `list_workspace_mfa_methods`: List MFA methods allowed for the workspace (`allowed_mfa_methods`, `passkey_as_mfa`); useful when guiding a user through the remote-mode MFA re-authentication flow
+- `get_workspace_notifications`: Get workspace notification settings (disconnection_notification, notification_channels)
+- `get_workspace_preferences`: Get workspace-wide preferences (timezone, locale, billing_email, etc.); workspace-global, not per-user
+- `update_workspace_notifications`: Update workspace notification settings (partial update)
+- `update_workspace_preferences`: Update workspace-wide preferences (partial update); changing `timezone` shifts the workspace's billing-usage aggregation boundary, and `billing_email`/`allowed_domains` are only accepted on SaaS deployments
+
+Access control and security settings are intentionally read-only here: the server gates writes to those governance-level settings behind a superuser session with fresh MFA, which a static API token (stdio mode) cannot satisfy, so they stay human-only via the web console.
```
Comment (junho226) [inline CLAUDE.md]: Suggestion: this rationale is SaaS-accurate but SaaS-centric. On-premise, `AccessControlSettingsViewSet` writes require only `IsAdminUser` + `HasTokenScope("workspaces")` and skip MFA (`initial()` bypasses it on-prem), so an admin API token *could* write these settings there. Omitting the write tool is still the right call — the stated reason just does not hold on-prem.

**Review by junho226 — DISMISSED** [review]
## Review — feat(workspace): add workspace settings management tools

Reviewed the full diff (+987 −16, 7 files) with the change checked out; 30/30 workspace tests pass and ruff is clean.

### Summary
Adds five read tools (`get_workspace_access_control`, `get_workspace_security`, `list_workspace_mfa_methods`, `get_workspace_notifications`, `get_workspace_preferences`) and two partial-update tools (`update_workspace_notifications`, `update_workspace_preferences`) over `/api/workspaces/`, plus five read-only `alpacon://workspace-settings/...` resources. Governance-level writes (access-control, security) are intentionally not exposed.

### Critical issues (0)
None. All new tools flow through the standard `@mcp_tool_handler` → `with_token_validation` path (identical to existing tools); endpoint paths are fixed literals with no user interpolation; PATCH bodies are built from a fixed named-param whitelist; authorization is enforced server-side.

### Warnings (2)
1. **`get_workspace_security` / `list_workspace_mfa_methods` are effectively JWT-only, undocumented.** Server-side auth classes carry no `APITokenAuthentication`, so a stdio-mode API token against a SaaS workspace hits a 403 that slips past the `_saas_only_security_404` guard. See inline note. Non-blocking.
2. **Update-tool list fields are full-array replacements but described as additive** (`notification_channels`, `enabled_extensions`, `allowed_domains`). An agent can silently drop existing entries. See inline notes. Non-blocking.

### Suggestions (1)
1. **CLAUDE.md rationale for omitting the access-control write tool is SaaS-centric** — on-premise, an admin API token could write those settings. The decision to omit the tool is still correct. See inline note.

### Highlights
- Auth/session/authorization fully inherited from `@mcp_tool_handler`; zero bespoke auth code.
- `_saas_only_security_404` keys on the error envelope + `status_code == 404`, so a success payload carrying `status_code: 404` is not misread — with two dedicated regression tests plus a 500-passthrough test.
- PATCH payloads built from a fixed named-param whitelist; no arbitrary-field injection surface.
- Strong coverage (30 tests) and accurate docs across README/api-reference/CHANGELOG/CLAUDE.md.

### Verdict: APPROVE
No critical issues; both warnings are documentation/UX hardening, not correctness or security defects.

**Comment by jisung-02** [comment]
Thanks for the thorough review, @junho226. All three points are addressed in 259d66c.

## Warning 1 — `get_workspace_security` / `list_workspace_mfa_methods` are JWT-only
Stacked `@require_jwt_auth` on both tools, so a static API token (stdio mode) now short-circuits with a clear message *before* the upstream call instead of surfacing the raw 403 that slipped past `_saas_only_security_404` (403 ≠ 404). This is the same precedent as the token-management tools you pointed to.

I also generalized the shared guard's error message — it previously said "API tokens cannot manage other API tokens", which is token-specific and would read wrong here. It now reads "requires JWT (OAuth/SSO) authentication and cannot be performed with a static API token", which fits both use cases. The existing token-tool tests only assert `'JWT' in message`, so they stay green.

The `@require_jwt_auth` guard and the `_saas_only_security_404` guard now layer cleanly: the guard handles the stdio API-token case, and the SaaS-only 404 branch still handles a JWT token hitting an on-premise deployment. Descriptions/docstrings updated to state the JWT-only + SaaS-only constraints, plus two new tests (`test_rejects_non_jwt_token` on each tool) asserting no request is sent.

## Warning 2 — list fields are full-array replacements, not additive
Reworded the descriptions and docstrings for `notification_channels`, `enabled_extensions`, and `allowed_domains` to state they REPLACE the whole list, with the read-merge-write guidance (read via the matching `get_*` tool, merge, then send the full list). Also documented that narrowing `enabled_extensions` 402s on non-enterprise plans. Mirrored the same notes in `docs/api-reference.md`.

## Suggestion — CLAUDE.md rationale was SaaS-centric
Corrected the note: the superuser+fresh-MFA gate only holds on SaaS. On-premise, `AccessControlSettingsViewSet` writes require only `IsAdminUser` + `HasTokenScope("workspaces")` and skip MFA, so an admin API token could technically write them there. The rationale now states that keeping governance-level settings human-only via the web console is the deliberate, safer default regardless of deployment — so omitting the write tool is still the right call.

Verification: full suite `928 passed`, and `ruff check` clean on the changed files.

**Review by junho226 — APPROVED** [review]
